borg/fuse.py

```diff
@@ -54,6 +55,7 @@ def __init__(self, key, repository, manifest, archive, cached_repo):
         self.pending_archives = {}
         self.accounted_chunks = {}
         self.cache = ItemCache()
+        self.data_cache = LRUCache(capacity=20, dispose=id)
```

Comment:
how about number_of_cores for the bound of this cache. and combine it with the explicit cache eviction i described above. would have a very reasonable memory footprint and handle parallel streams w/o troubles.